tests/eosio.rentbw_tests.cpp

```diff
@@ -94,39 +94,138 @@ struct rentbw_tester : eosio_system_tester
       create_accounts_with_resources({N(eosio.reserv)});
    }
 
+   asset calc_total_fee(const rentbw_state& state, int64_t net_frac, int64_t cpu_frac) 
+   {
+      auto net_util = __int128_t(net_frac) * state.net.weight / rentbw_frac;
+      auto net_fee = calc_rentbw_fee(state.net, net_util);
+      auto cpu_util = __int128_t(cpu_frac) * state.cpu.weight / rentbw_frac;
+      auto cpu_fee = calc_rentbw_fee(state.cpu, cpu_util);
+      
+      ilog("net_fee: ${net_fee} cpu_fee: ${cpu_fee}",("net_fee",net_fee)("cpu_fee",cpu_fee));
+      auto fee = asset(net_fee + cpu_fee, symbol{CORE_SYM});
+      return fee;
+   }
+
+   asset calc_both_fees(const rentbw_state& state, int64_t net_frac, int64_t cpu_frac) 
+   {
+      auto net_util = __int128_t(net_frac) * state.net.weight / rentbw_frac;
+      auto net_fee = calc_rentbw_fee(state.net, net_util);
+      auto cpu_util = __int128_t(cpu_frac) * state.cpu.weight / rentbw_frac;
+      auto cpu_fee = calc_rentbw_fee(state.cpu, cpu_util);
+      
+      asset fee = core_sym::from_string("0.0000");
+      if (net_fee > 0 && cpu_fee > 0) {
+         ilog("net_fee: ${net_fee} cpu_fee: ${cpu_fee}",("net_fee",net_fee)("cpu_fee",cpu_fee));
+         fee = asset(net_fee + cpu_fee, symbol{CORE_SYM});
+      }
+      return fee;
+   }
+
+   int64_t calc_rentbw_fee(const rentbw_state_resource& state, int64_t utilization_increase) {
+      if( utilization_increase <= 0 ) return 0;
+
+      // Let p(u) = price as a function of the utilization fraction u which is defined for u in [0.0, 1.0].
+      // Let f(u) = integral of the price function p(x) from x = 0.0 to x = u, again defined for u in [0.0, 1.0].
+
+      // In particular we choose f(u) = min_price * u + ((max_price - min_price) / exponent) * (u ^ exponent).
+      // And so p(u) = min_price + (max_price - min_price) * (u ^ (exponent - 1.0)).
+
+      // Returns f(double(end_utilization)/state.weight) - f(double(start_utilization)/state.weight) which is equivalent to
+      // the integral of p(x) from x = double(start_utilization)/state.weight to x = double(end_utilization)/state.weight.
+      // @pre 0 <= start_utilization <= end_utilization <= state.weight
+      auto price_integral_delta = [&state](int64_t start_utilization, int64_t end_utilization) -> double {
+         double coefficient = (state.max_price.get_amount() - state.min_price.get_amount()) / state.exponent;
+         double start_u     = double(start_utilization) / state.weight;
+         double end_u       = double(end_utilization) / state.weight;
+         return state.min_price.get_amount() * end_u - state.min_price.get_amount() * start_u +
+                  coefficient * std::pow(end_u, state.exponent) - coefficient * std::pow(start_u, state.exponent);
```

Comment:
It can be shortened to:
return state.min_price.get_amount() * (end_u - start_u) + coefficient * (std::pow(end_u, state.exponent) - std::pow(start_u, state.exponent))

Reply:
This code is essentially a copy-paste from `rentbw.cpp` so I left it as is.